src/p5.dance.js

```diff
@@ -178,7 +180,11 @@ module.exports = class DanceParty {
     if (this.recordReplayLog_) {
       replayLog.reset();
     }
-    this.playSound_({url: METADATA[this.getSelectedSong_()].file, callback: () => {this.songStartTime_ = new Date()}});
+    this.playSound_({
+      url: METADATA[this.getSelectedSong_()].file,
+      callback: () => {this.songStartTime_ = new Date()},
+      onEnded: () => {this.reset(); this.toggleRunButton_();}
+    });
```

Comment:
It makes sense to call `this.reset()` here, but I'm not sure we want to rely on this class to notify the caller that the song completed (seems like dependencies are going the wrong way).

Is there a way to do this with a pub/sub pattern: attaching multiple listeners to the `playSound` API, so the caller can get notified directly when the song ends?